src/CMakeLists.txt

```diff
@@ -125,10 +125,10 @@ if(ENABLE_SIO)
 
   PODIO_ADD_LIB_AND_DICT(podioSioIO "${sio_headers}" "${sio_sources}" sio_selection.xml)
   target_link_libraries(podioSioIO PUBLIC podio::podio SIO::sio ${CMAKE_DL_LIBS} ${PODIO_FS_LIBS})
+  target_compile_definitions(podioSioIO PUBLIC PODIO_ENABLE_SIO=1)
```

Comment:
Does it need to have a value?

Reply:
Yes, there are places where we depend on the value, e.g.
https://github.com/AIDASoft/podio/blob/9cd03731dfe4b4c37eac34e6364411bb55be940a/src/IOHelpers.cc#L26-L28

I think it is the cleaner solution to define this to a non-false value rather then depending on whether it is simply defined. I could also change the usage in the unittests, to reflect that.

Additionally, from the [cmake documentation](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html) it seems we can be explicit here rather than implicit without changing things:
> Definitions may optionally have values:
> 
> ```cmake
> target_compile_definitions(foo PUBLIC FOO=1)
> ```
> Note that many compilers treat -DFOO as equivalent to -DFOO=1, but other tools may not recognize this in all circumstances